packages/core/admin/admin/src/content-manager/components/BlocksInput/Blocks/List.tsx

```diff
@@ -29,26 +29,42 @@ const listStyle = css`
   }
 `;
 
-const Orderedlist = styled.ol`
-  list-style-type: decimal;
+const Orderedlist = styled.ol<{ listStyleType: CSSProperties['listStyleType'] }>`
+  list-style-type: ${(props) => props.listStyleType};
   ${listStyle}
 `;
 
-const Unorderedlist = styled.ul`
-  list-style-type: disc;
+const Unorderedlist = styled.ul<{ listStyleType: CSSProperties['listStyleType'] }>`
+  list-style-type: ${(props) => props.listStyleType};
   ${listStyle}
 `;
 
+const orderedStyles = ['decimal', 'lower-alpha', 'upper-roman'];
+const unorderedStyles = ['disc', 'circle', 'square'];
+
 const List = ({ attributes, children, element }: RenderElementProps) => {
   if (!isListNode(element)) {
     return null;
   }
 
+  // Determine next style based on the current style
+  const listStyles = element.format === 'ordered' ? orderedStyles : unorderedStyles;
+  const nextIndex = (element.indentLevel || 0) % listStyles!.length;
```

Comment:
It took me a while to work this out, basically we allow infinite nesting so we use the `%` symbol to "wrap" round the styles if we nest four levels, for example. Might be worth writing that as a comment as that behaviour isn't immeditely obvious.